plugins/schedule/base/lib/model/ScheduleEvent.php

```diff
@@ -264,5 +264,23 @@ public function getResourceIdsForIndex()
 	
 		return self::RESOURCES_INDEXED_FIELD_PREFIX . $this->getPartnerId() . " " .  implode(' ', $index);
 	}
-	
-} // ScheduleEvent
+
+	public function getTemplateEntryCategoriesIdsForIndex()
+	{
+		return '';
+	}
+
+	public function getSystemNamesForIndex()
+	{
+		$eventResources = ScheduleEventResourcePeer::retrieveByEventId($this->getId());
+
+		$system_names = array();
+		foreach($eventResources as $eventResource)
+		{
+			$resource = ScheduleResourcePeer::retrieveByPK($eventResource->getResourceId());
+			KalturaLog::info($resource->getSystemName());
+			$system_names[] = $resource->getSystemName;
+		}
+		return implode(' ', $system_names);

```

Comment:
Since system name may contain spaces and special characters, I suggest to md5 them during indexing and during search.